src/js/libcs/OpDrawing.js

```diff
@@ -106,6 +106,183 @@ evaluator.fillcircle$2 = function(args, modifs) {
     return eval_helper.drawcircle(args, modifs, "F");
 };
 
+evaluator.drawarc$3 = function(args, modifs) {
+    return eval_helper.drawarc(args, modifs, "D");
+};
+
+evaluator.fillarc$3 = function(args, modifs) {
+    return eval_helper.drawarc(args, modifs, "F");
+};
+
+
+eval_helper.drawarc = function(args, modifs, df) {
+    var v1 = evaluateAndVal(args[0]);
+    var v2 = evaluateAndVal(args[1]);
+    var v3 = evaluateAndVal(args[2]);
+
+    var a = List.normalizeZ(csgeo.csnames[args[0].name].homog);
+    var b = List.normalizeZ(csgeo.csnames[args[1].name].homog);
+    var c = List.normalizeZ(csgeo.csnames[args[2].name].homog);
+    var abcdet = List.det3(a, b, c);
```

Comment:
I wonder what happens if one of the points is at infinity. If `b* is atinfinity, then you *should* get a pair of rays, i.e. what you get in the collinear case if `b` is outside the segment `ac`. The direction of `b` is irrelevant. If `a` is anywhere at infinity, then you'd get a single ray starting at `c` and passing through `b`, and conversely if `c` is at infinity. Long story short: it might be better to move that normalization to somewhere where these special cases have already been excluded. If you want to you can leave that for a later commit, though.